src/scrapers/yahav.ts

```diff
@@ -60,10 +60,9 @@ function getPossibleLoginResults(page: Page): PossibleLoginResults {
 }
 
 async function getAccountID(page: Page) {
-  const selectedSnifAccount = await page.$eval(`${ACCOUNT_ID_SELECTOR}`, (option) => {
-    return (option as HTMLElement).innerText;
+  const selectedSnifAccount = await page.$eval(ACCOUNT_ID_SELECTOR, (e: Element) => {
+    return e.textContent ?? '';
```

Comment:
If you return an empty string you didn't handle the unexpected behavior, you just hide it and it will be thrown to another place.

Please handle the possibility of an empty string where this function is called.

Reply:
@baruchiro I understand your point. The thing is that once you pass the login, there is no way the account ID is empty... except if the page was changed, that raises an error (and that's why I found this...)
WDYT? Raise another error here just in case?

Reply:
like this?

```
async function getAccountID(page: Page): Promise<string> {
  try {
    // Attempt to extract the account ID using the selector
    const selectedSnifAccount = await page.$eval(ACCOUNT_ID_SELECTOR, (element: Element) => {
      return element.textContent as string;
    });
    return selectedSnifAccount;
  } catch (error) {
    // Handle the case where the selector is outdated or the element is not found
    const errorMessage = error instanceof Error ? error.message : String(error);
    throw new Error(`Failed to retrieve account ID. Possible outdated selector. ${errorMessage}`);
  }
}
```

Reply:
If you don't expect the string to be empty, ignore that case completely.

If the string is empty sometimes, the code will crash because of a page change.

You may be surprised, but this is the expected behavior- if the page was changed and something unexpected was received, then yes, crash it.